### ISSUE TITLE
Build Error

### ISSUE THREAD

**Issue by ghost** [polatory/polatory#114]
Hi. I am trying to build Polatory on win64 and I get this error after I'm trying to .\run init-vcpkg
```
Starting package 11/1: intel-mkl:x64-windows
Building package intel-mkl[core]:x64-windows...
Could not locate cached archive: C:\Users\Administrator\AppData\Local\vcpkg\archives\80\8042fd732ac8c52e31cd068c4c7b09a0ad0eb54c.zip
CMake Error at ports/intel-mkl/portfile.cmake:15 (message):
  Could not find MKL.  Before continuing, please download and install MKL
  (20200000 or higher) from:

      https://registrationcenter.intel.com/en/products/download/3178/



  Also ensure vcpkg has been rebuilt with the latest version (v0.0.104 or
  later)
Call Stack (most recent call first):
  scripts/ports.cmake:135 (include)


Error: Building package intel-mkl:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: intel-mkl:x64-windows
  Vcpkg version: 2020.06.15-nohash
```

I can't manage to download the old version of intel mkl. Do I really need 2019.5 one or the latest is OK it's just something else is going wrong on my machine? thank you.

**Comment by nakedtiger666** [comment]
Okay. so I've managed to investigate the issue and turned out the path of the mkl is changed in the 2021 version of the oneAPI. I did the following changes in the polatory\vcpkg\ports\intel-mkl\portfile.cmake of the vcpkg and it worked. I updated the path and disabled the version match check (for some reason it didn't compare correctly even though the version is higher).

```
# Due to the complexity involved, this package doesn't install MKL. It instead verifies that MKL is installed.
# Other packages can depend on this package to declare a dependency on MKL.
# If this package is installed, we assume that MKL is properly installed.

set(VCPKG_POLICY_EMPTY_PACKAGE enabled)

set(MKL_REQUIRED_VERSION "20180000")

set(ProgramFilesx86 "ProgramFiles(x86)")
set(INTEL_ROOT $ENV{${ProgramFilesx86}}/Intel/oneAPI/mkl/2021.1.1)

find_path(MKL_ROOT include/mkl.h PATHS ${INTEL_ROOT} DOC "Folder contains MKL")

if (MKL_ROOT STREQUAL "MKL_ROOT-NOTFOUND")
    message(FATAL_ERROR "Could not find MKL. MKL root ${MKLROOT}")
endif()

# file(STRINGS ${MKL_ROOT}/include/mkl_version.h MKL_VERSION_DEFINITION REGEX "__INTEL_MKL((_MINOR)|(_UPDATE))?__")
# string(REGEX MATCHALL "([0-9]+)" MKL_VERSION ${MKL_VERSION_DEFINITION})
# list(GET MKL_VERSION 0 MKL_VERSION_MAJOR)
# list(GET MKL_VERSION 1 MKL_VERSION_MINOR)
# list(GET MKL_VERSION 2 MKL_VERSION_UPDATE)

file(STRINGS ${MKL_ROOT}/include/mkl_version.h MKL_VERSION_DEFINITION REGEX "INTEL_MKL_VERSION")
string(REGEX MATCH "([0-9]+)" MKL_VERSION ${MKL_VERSION_DEFINITION})

#if (MKL_VERSION LESS MKL_REQUIRED_VERSION)
#    message(FATAL_ERROR "MKL ${MKL_VERSION} is found but ${MKL_REQUIRED_VERSION} is required. Please download and install a more recent version of MKL from:"
#                        "\n    https://registrationcenter.intel.com/en/products/download/3178/\n")
#endif()
```

**Comment by nakedtiger666** [comment]
Also in order to run build I also updated root CMakeLists.txt. Updated the dlls paths and actually I just copied the mkl from the new location to the manually created old required location, because looks like some code still require it. Obviously it is a very rough hack to make it work. Looking forward for the proper resolution of the issue! 

```
   if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
        message(FATAL_ERROR "Not supported compiler.")
    endif()

    set(_PF86 "ProgramFiles(x86)")
    file(TO_CMAKE_PATH "$ENV{${_PF86}}" _PROG_FILES_X86)

    if("$ENV{APPVEYOR}" STREQUAL "True")
        file(TO_CMAKE_PATH "$ENV{CONDA_DIR}" CONDA_DIR)

        set(MKL_ROOT "${CONDA_DIR}/Library")
        set(MKL_DLL_DIR "${MKL_ROOT}/bin")
        set(MKL_INCLUDE_DIR "${MKL_ROOT}/include")
        set(MKL_LIB_DIR "${MKL_ROOT}/lib")
    else()
        set(MKL_ROOT "${_PROG_FILES_X86}/Intel/oneAPI/mkl/2021.1.1" CACHE PATH "Path to mkl")
        set(MKL_DLL_DIR "${MKL_ROOT}/redist/intel64/")
        set(MKL_INCLUDE_DIR "${MKL_ROOT}/include")
        set(MKL_LIB_DIR "${MKL_ROOT}/lib/intel64")
    endif()

    if(USE_PREBUILT_BOOST)
        if("${Boost_INCLUDE_DIR}" STREQUAL "${VCPKG_INCLUDE_DIR}")
            message(FATAL_ERROR "Boost libraries from vcpkg are detected. Turn off USE_PREBUILT_BOOST.")
        endif()
    endif()

    set(POLATORY_DLLS
        ${MKL_DLL_DIR}//mkl_avx.1.dll
        ${MKL_DLL_DIR}/mkl_avx2.1.dll
        ${MKL_DLL_DIR}/mkl_avx512.1.dll
        ${MKL_DLL_DIR}/mkl_core.1.dll
        ${MKL_DLL_DIR}/mkl_def.1.dll
        ${MKL_DLL_DIR}/mkl_mc.1.dll
        ${MKL_DLL_DIR}/mkl_mc3.1.dll
        ${MKL_DLL_DIR}/mkl_sequential.1.dll
    )
```

**Comment by namihisa-gsinet** [comment]
I consider modifying CMakeLists.txt or README to enable the build of a new version of intel mkl.

**Comment by nakedtiger666** [comment]
thanks!

**Comment by unageek** [comment]
Fixed in #122.
